File: TP1 - C++/data/models/consultation.cpp
#include "../../domain/entities/consultation.hpp"

Consultation::Consultation(string id, Patient patient, string date, string description, PaymentConsultation paymentConsultation):
    id(id),
    patient(patient),
    date(date),
    description(description),
    paymentConsultation(paymentConsultation){};

Consultation::~Consultation(){};

string Consultation::getId(){
    return this->id;
}

Patient Consultation::getPatient(){
    return this->patient;
}

string Consultation::getDate(){
    return this->date;
}

string Consultation::getDescription(){
    return this->description;
}

PaymentConsultation Consultation::getPaymentConsultation(){
    return this->paymentConsultation;
}

void Consultation::setPatient(Patient patient){
    this->patient = patient;
}

void Consultation::setDate(string date){
    this->date = date;
}

void Consultation::setDescription(string description){
    this->description = description;
}

void Consultation::setPaymentConsultation(PaymentConsultation paymentConsultation){
    this->paymentConsultation = paymentConsultation;
}

File: TP2 - Java/src/domain/entities/Consultation.java
package domain.entities;

public class Consultation {
    private final String id;
    private Patient patient;
    private String date;
    private String description;
    private PaymentConsultation paymentConsultation;

    public Consultation(String id, Patient patient, String date, String description, PaymentConsultation paymentConsultation) {
        this.id = id;
        this.patient = patient;
        this.date = date;
        this.description = description;
        this.paymentConsultation = paymentConsultation;
    }

    public String getId() {
        return id;
    }

    public Patient getPatient() {
        return patient;
    }

    public void setPatient(Patient patient) {
        this.patient = patient;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public PaymentConsultation getPaymentConsultation() {
        return paymentConsultation;
    }

    public void setPaymentConsultation(PaymentConsultation paymentConsultation) {
        this.paymentConsultation = paymentConsultation;
    }
}
